### ISSUE TITLE
Crash while exploring 

### ISSUE THREAD

**Issue by hlangeveld** [anegostudios/VintageStory-Issues#968]
**Game Version:** 1.15.0-pre.2
**Platform:** Windows
**Modded:** No

### Description
Running around and exploring. Suddenly the game froze, then crashed.

Update: I was able to recover the world. Nothing special about the world in spectator mode, except for a large number of rabbits/hares around the place of the crash. Three or four within just a couple of blocks.
![image](https://user-images.githubusercontent.com/1230736/120908160-defd3780-c667-11eb-9224-69676028b8cf.png)

### How to reproduce
1. Unknown - perhaps my gameworld was corrupted earlier?
2. Note: The game world was originally created in 1.15.0-pre1 and upgraded to -pre2
3. Everytime I open this world I get a prompt about object mapping, and if I want to continue.
    I let it create a backup and update the object mapping, but the prompt keeps popping up again.
4. I did encounter the plank-stacking bug earlier in this game, and had to repair my savegame for that.

### Expected behavior
If not obvious, what did you expect to happen?

### Screenshots
Attach a screenshot if it helps explain the issue / shows a visual bug.

### Logs
```
Paste crash here.

Running on 64 bit Windows with 16 GB RAM 
Version: v1.15.0-pre.2 (Unstable)06/06/2021 00:24:36: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Common.WorldChunk.GetLocalBlockAtBlockPos(IWorldAccessor world, BlockPos position)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayer(SnowAccumSnapshot sumsnapshot, Boolean ignoreOldAccum, IServerMapChunk mc, Vec2i chunkPos, IWorldChunk[] chunksCol)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.GetSnowUpdate(WeatherSimulationRegion simregion, IServerMapChunk mc, Vec2i chunkPos, IWorldChunk[] chunksCol)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayerOffThread(WeatherSimulationRegion simregion, IServerMapChunk mc, Vec2i chunkPos)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.onThreadTick()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 06/06/2021 00:24:48, Site = , Source = Windows Error Reporting, Message = Fault bucket 1888892606090427954, type 5
Event Name: CLR20r3
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.0.0
P3: 60b27037
P4: VintagestoryLib
P5: 1.15.0.0
P6: 60b27036
P7: 1419
P8: 2e
P9: System.NullReferenceException
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER44F0.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER5452.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER54A1.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER54AF.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER54EF.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_a9124d9692b5fb7563bc7177d11c97f18f76ed_387ef1c3_5b1965fb-d425-43f0-8b89-77ffe7e71f9a

Analysis symbol: 
Rechecking for solution: 0
Report Id: e8948f2d-e436-4717-ac86-2ef0d369ac16
Report Status: 268435456
Hashed bucket: 35590c4783bbafe59a36b1d144e1be32
Cab Guid: 0 }
--------------
{ TimeGenerated = 06/06/2021 00:24:41, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.15.0.0, time stamp: 0x60b27037
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x00007ffad96b5d89
Faulting process ID: 0x4cd0
Faulting application start time: 0x01d75a52d2c6781f
Faulting application path: C:\Extra\Vintagestory\1-15\Vintagestory.exe
Faulting module path: unknown
Report ID: e8948f2d-e436-4717-ac86-2ef0d369ac16
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 06/06/2021 00:24:39, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Vintagestory.Common.WorldChunk.GetLocalBlockAtBlockPos(Vintagestory.API.Common.IWorldAccessor, Vintagestory.API.MathTools.BlockPos)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayer(Vintagestory.GameContent.SnowAccumSnapshot, Boolean, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i, Vintagestory.API.Common.IWorldChunk[])
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.GetSnowUpdate(Vintagestory.GameContent.WeatherSimulationRegion, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i, Vintagestory.API.Common.IWorldChunk[])
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayerOffThread(Vintagestory.GameContent.WeatherSimulationRegion, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.onThreadTick()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart()

 }

In the game's Logs folder you may be able to find more information
inside client-main.txt and server-main.txt. Alternatively, you may use
gist.github.com to upload logs, linking them here, if they're too large.

Note that the crash log contains all crashes that have happened so far.
You likely only want to report the latest one.
```

**Comment by Craluminum2413** [comment]
Duplicate of #908

**Comment by copygirl** [comment]
> Everytime I open this world I get a prompt about object mapping, and if I want to continue.
> I let it create a backup and update the object mapping, but the prompt keeps popping up again.

If you uncheck the backup option it will correctly apply the mappings and rejoin the world.

**Comment by hlangeveld** [comment]
> > Everytime I open this world I get a prompt about object mapping, and if I want to continue.
> > I let it create a backup and update the object mapping, but the prompt keeps popping up again.
> 
> If you uncheck the backup option it will correctly apply the mappings and rejoin the world.

Oops, *that* results in an immediate crash :-)

```
Running on 64 bit Windows with 16 GB RAM 
Version: v1.15.0-pre.2 (Unstable)06/06/2021 13:58:06: Critical error occurred
System.Exception: GL_INVALID_OPERATION error generated. Buffer name does not refer to an buffer object generated by OpenGL.
   at Vintagestory.Client.NoObf.ClientPlatformWindows.DebugCallback(DebugSource source, DebugType type, Int32 id, DebugSeverity severity, Int32 length, IntPtr message, IntPtr userParam)
   at OpenTK.Graphics.OpenGL.GL.BindBuffer(BufferTarget target, Int32 buffer)
   at Vintagestory.Client.NoObf.ClientPlatformWindows.UpdateMesh(MeshRef modelRef, MeshData data)
   at Vintagestory.API.Client.MeshDataPool.InsertAt(ICoreClientAPI capi, MeshData modeldata, Vec3i modelOrigin, Sphere frustumCullSphere, Int32 indexPosition, Int32 vertexPosition, Int32 listPosition)
   at Vintagestory.API.Client.MeshDataPool.TryAppend(ICoreClientAPI capi, MeshData modeldata, Vec3i modelOrigin, Sphere frustumCullSphere)
   at Vintagestory.API.Client.MeshDataPool.TryAdd(ICoreClientAPI capi, MeshData modeldata, Vec3i modelOrigin, Sphere frustumCullSphere)
   at Vintagestory.API.Client.MeshDataPoolManager.AddModel(MeshData modeldata, Vec3i modelOrigin, Sphere frustumCullSphere)
   at Vintagestory.Client.NoObf.TesselatedChunkPart.AddToPools(ChunkRenderer cr, List`1 locations, Vec3i chunkOrigin, Sphere boundingSphere, BoolRef[] cullVisible)
   at Vintagestory.Client.NoObf.TesselatedChunk.AddCenterToPools(ChunkRenderer chunkRenderer, Vec3i chunkOrigin, Sphere boundingSphere, ClientChunk hostChunk)
   at Vintagestory.Client.NoObf.ChunkRenderer.AddTesselatedChunk(TesselatedChunk tesschunk, ClientChunk hostChunk)
   at Vintagestory.Client.NoObf.ChunkTesselatorManager.OnFinalizeFrame(Single dt)
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt)
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt)
   at _xArlFw5lpThzIY0BN36UerAlHzU._14ww1iwrPaPwPpAZqZCUdf3d4Sm(Single )
   at _uFwaliVbJcYmzlP3dwic7HSHMr2._EuQCmN4cQf8FBHYrlJxd8EhIdNnA(Single )
   at _uFwaliVbJcYmzlP3dwic7HSHMr2._HlDCi3DfsJHaHE4jniBarrjFlHLB(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _EKato5MKvGk9ELCMKyl4BGVO7pe._ninJWCWaWyMG9ZSA1Eyytms1G5F(_BjQ4x8rTRChySIeXrK1HK1YSqec , String[] )
   at _btwtpE4G1MfBqD34njKFCvBcSRd._ninJWCWaWyMG9ZSA1Eyytms1G5F(ThreadStart )
```

Just parking that trace here for now.

**Comment by copygirl** [comment]
Is that a one-time crash, does it happen when you rejoin the world, or is it even reproducible?

**Comment by hlangeveld** [comment]
> Is that a one-time crash, does it happen when you rejoin the world, or is it even reproducible?

One time only, at the moment where I give the ok for the remapping. World crashes, I restart, and all problems are solved.

Looks testable, by creating a world in 1.15.0-pre1 and loading it in 1.15.0-pre2 (which I might try).  We can copy this trace to a new bug, and copy your instructions there, as they *do* work.

**Comment by radfast** [comment]
This crash:

    System.NullReferenceException: Object reference not set to an instance of an object.
       at Vintagestory.Common.WorldChunk.GetLocalBlockAtBlockPos(IWorldAccessor world, BlockPos position)
       at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayer(SnowAccumSnapshot sumsnapshot, Boolean ignoreOldAccum, IServerMapChunk mc, Vec2i chunkPos, IWorldChunk[] chunksCol)

This one is fixed in `1.15-pre.3` (see #News channel on Discord).  If you still have the same issue please re-report it and we should give it high priority.

Any other issues please open a new issue report :)   The remapping / object mapping / meshdatapool points you have been discussing more recently are probably not related to the original SnowAccum crash.

**Comment by hlangeveld** [comment]
> This one is fixed in `1.15-pre.3` (see #News channel on Discord). If you still have the same issue please re-report it and we should give it high priority.

Confirmed
